packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts

```diff
@@ -163,7 +172,9 @@ export class EcsTask implements events.IRuleTarget {
     const taskDefinitionArn = this.taskDefinition.taskDefinitionArn;
 
     const subnetSelection = this.props.subnetSelection || { subnetType: ec2.SubnetType.PRIVATE };
-    const assignPublicIp = subnetSelection.subnetType === ec2.SubnetType.PUBLIC ? 'ENABLED' : 'DISABLED';
+    const assignPublicIp = this.props.assignPublicIp ??
+      subnetSelection.subnetType === ec2.SubnetType.PUBLIC ? 'ENABLED' : 'DISABLED';
```

Comment:
We can't know this here, because you might be selecting a subnet by name or some other criterion. Looking at the `SubnetType` here is pretty much wrong.

If anywhere, we should probably do it after we do `selectSubnets`, because the result object of that will contain a field indicating whether we selected any public subnets.

But also -- what happens if we leave this value out? Would the default perhaps be correct already?

---------

> launch type must also be FARGATE in order for ENABLED to have impact. Should we warn there?

That's a good idea. Yes, let's.